src/server/ua_server_binary.c

```diff
@@ -54,6 +54,7 @@ static void processHEL(UA_Connection *connection, const UA_ByteString *msg, UA_U
     UA_TcpMessageHeader_encodeBinary(&ackHeader, &ack_msg, &tmpPos);
     UA_TcpAcknowledgeMessage_encodeBinary(&ackMessage, &ack_msg, &tmpPos);
     UA_ByteStringArray answer_buf = { .stringsSize = 1, .strings = &ack_msg };
+    // the string is freed internall in the (asynchronous) write
```

Comment:
The string does not have to be freed at all.
UA_alloca just decreases (the stack grows from the top) the esp register and returns a pointer.
When the function returns, esp is reset to the value it had before the function was called.

This works for the ACK message because its size is small enough.
We would have to change this asynchronous writes...